### PR TITLE
Funnel improvements IV

### LINKED ISSUE #5201: Overlapping funnel text on smaller screen size
## Bug description
In the funnel panel the text is overlapping when I have a long name and a smaller screen size
image

## Expected behavior
Maybe the name should be shortened to ... or we could wrap it - no preference but this looks a little messy.

## How to reproduce

1. Go to funnel with a step other than the last with a long name (perhaps 50 characters)
2. Reduce horizontal width of screen

## Environment

- [x] PostHog Cloud

## Additional context
I was using safari.

### PR DESCRIPTION
**Pull request by paolodamico**
## Changes

More improvements to funnels. In this instance:
- Fixes bug in which funnels where not rendered on dashboards (https://github.com/PostHog/posthog/issues/5171#issuecomment-882504709)
- Refactors how the new funnel components get rendered to not rely on `FunnelViz` component anymore.
- Fixes the overflow problem with long event names (in both dashboards and regular insights view) [closes #5201]

## Checklist

- [ ] All querysets/queries filter by Organization, by Team, and by User
- [ ] Django backend tests
- [ ] Jest frontend tests
- [ ] Cypress end-to-end tests
- [ ] Migrations are safe to run at scale (e.g. PostHog Cloud) – present proof if not obvious
- [ ] New/changed UI is decent on smartphones (viewport width around 360px)

### CODE REVIEW

**Comment by paolodamico** [comment]
Only blocker to start cloud release of new funnels feature.

File: frontend/src/scenes/funnels/funnelLogic.ts [hunk 1]
```diff
@@ -212,12 +212,12 @@ export const funnelLogic = kea<funnelLogicType>({
                         ])
                         breakpoint()
                         insightLogic.actions.endQuery(queryId, ViewType.FUNNELS, result.last_refresh)
-
                         return { results: result.result, timeConversionResults }
                     } catch (e) {
                         if (!isBreakpoint(e)) {
                             insightLogic.actions.endQuery(queryId, ViewType.FUNNELS, null, e)
                         }
+                        console.error(e)
```
Comment (mariusandra) [inline frontend/src/scenes/funnels/funnelLogic.ts:220]: debug log?

**Review by mariusandra — COMMENTED** [review]
Very nice! Found some additional issues with the dashboard funnels though:

- don't work with [the different colors](https://github.com/PostHog/posthog/issues/5171#issuecomment-881627562)
- don't save the "top to bottom" option

image

- the tooltip is also off:

image

All of this was with postgres. Still trying to get ClickHouse running on the M1. Got [pretty far](https://github.com/PostHog/posthog/pull/5215) though!

File: frontend/src/scenes/funnels/funnelLogic.ts (hunk 1 above)
Comment (paolodamico) [reply frontend/src/scenes/funnels/funnelLogic.ts:220]: This was actually intended as to not swallow the errors

**Comment by paolodamico** [comment]
I'm aware of the outstanding dashboard bugs, will address in a separate PR.

**Review by mariusandra — APPROVED** [review]

**Pull request merged**